src/zitadel.rs

```diff
@@ -373,7 +373,11 @@ impl User {
 		let preferred_username = read_entry(&entry, &config.ldap.attributes.preferred_username)?;
 		let email = read_entry(&entry, &config.ldap.attributes.email)?;
 		let user_id = read_entry(&entry, &config.ldap.attributes.user_id)?;
-		let phone = read_entry(&entry, &config.ldap.attributes.phone)?;
+		let phone = if let Some(phone_attribute) = config.ldap.attributes.phone {
+			Some(read_entry(&entry, &phone_attribute)?);
+		} else {
+			None;
+		};
```

Comment:
mb smth like 
```rust
config.ldap.attributes.phone.map(|phone_attribute| read_entry(&entry, &phone_attribute)).transpose()?
```
?

Reply:
Another suggestion just for readability: bind `let attributes = config.ldap.attributes;` shorter `attributes` in `read_entry` calls

Reply:
Yeah, this was done to get it over the line quickly while we were actively testing, didn't want to think through clever option handling.

The attribute binding is a nice idea too. I wonder if something more generic would be possible in general, but I guess we'd need to be able to iterate over struct field names.

Reply:
Yeah, we could make something like 
```rust 
enum Attributes {
  PhoneNumber,
  FirstName,
  ...
}

type LdapAttributesMapping = HashMap<Attributes, String>
```
Then we could iterate over it, but then all attributes would become optional on type level, so I think it's fine as is